third_party/HugeCTR/gpu_cache/src/nv_gpu_cache.cu

```diff
@@ -30,37 +31,40 @@ __forceinline__ __device__ long atomicAdd(long* address, long val) {
   return (long)atomicAdd((unsigned long long*)address, (unsigned long long)val);
 }
 
-__forceinline__ __device__ long long atomicAdd(
-    long long* address, long long val) {
-  return (long long)atomicAdd(
-      (unsigned long long*)address, (unsigned long long)val);
+__forceinline__ __device__ long long atomicAdd(long long* address, long long val) {
+  return (long long)atomicAdd((unsigned long long*)address, (unsigned long long)val);
 }
 
+// Defining this causes this error:
+// /.../dgl/third_party/HugeCTR/gpu_cache/src/nv_gpu_cache.cu:33:42: error: redefinition of 'atomicAdd'
+//    33 | __forceinline__ __device__ unsigned long atomicAdd(unsigned long* address, unsigned long val) {
+//       |                                          ^
+// /opt/rocm-6.2.3/lib/llvm/bin/../../../include/hip/amd_detail/amd_hip_atomic.h:240:15: note: previous definition is here
+//   240 | unsigned long atomicAdd(unsigned long* address, unsigned long val) {
+//       |               ^
 #ifndef DGL_USE_HIP
-__forceinline__ __device__ unsigned long atomicAdd(
-    unsigned long* address, unsigned long val) {
-  return (unsigned long)atomicAdd(
-      (unsigned long long*)address, (unsigned long long)val);
+__forceinline__ __device__ unsigned long atomicAdd(unsigned long* address, unsigned long val) {
+  return (unsigned long)atomicAdd((unsigned long long*)address, (unsigned long long)val);
 }
 #endif
 
 namespace gpu_cache {
 
 #ifdef LIBCUDACXX_VERSION
 template <int warp_size>
-__forceinline__ __device__ void warp_tile_copy(
-    const size_t lane_idx, const size_t emb_vec_size_in_float, float* d_dst,
-    const float* d_src) {
+__forceinline__ __device__ void warp_tile_copy(const size_t lane_idx,
+                                               const size_t emb_vec_size_in_float, float* d_dst,
+                                               const float* d_src) {
```

Comment:
I think we should be trying to minimize diffs with upstream. Applying arbitrary formatting fixes does not seem worth it.

Reply:
Hmm the diff to upstream is actually small and all non-formatting changes https://github.com/dmlc/dgl/compare/master...ROCm:dgl:feat/jamesETsmith/graphbolt_macros#diff-4069a673e3582b44924c3bf91a4ead92d82e226b67b771c383664aada5bb4975

So maybe they reformatted recently. I can look into it more later